index.js

```diff
@@ -282,7 +282,7 @@ is.arraylike = function (value) {
   return !!value && !is.bool(value)
     && owns.call(value, 'length')
     && isFinite(value.length)
-    && is.number(value.length)
+    && is.integer(value.length)
```

Comment:
Thanks for your explanation. 
> https://tc39.github.io/ecma262/#sec-tolength
> The abstract operation ToLength converts argument to an integer suitable for use as the length of an array-like object. It performs the following steps:
> 1. Let len be ? ToInteger(argument).
> 2. If len ≤ +0, return +0.
> 3. Return min(len, 2^53-1).

I'm still confused about this. Since the spec coerces negative numbers to +0 and infinite numbers to `min(len, 2^53-1)` too. Should the `is.arraylike` be changed to the below?
```javascript
is.arraylike = function (value) {
    return !!value && !is.bool(value)
    && owns.call(value, 'length')
    // && isFinite(value.length)
    && is.number(value.length);
    // && value.length >= 0;
};
```

Here are some other `isArrayLike` ...Some checked the integer,some not...Really confused...
- loadash
```javascript
function isArrayLike(value) {
  return value != null && typeof value != 'function' && isLength(value.length)
}
function isLength(value) {
  return typeof value == 'number' &&
    value > -1 && value % 1 == 0 && value <= MAX_SAFE_INTEGER // checked
}
```
- `jquery` 
```javascript
function isArrayLike( obj ) {
	var length = !!obj && "length" in obj && obj.length,
		type = jQuery.type( obj );
	if ( jQuery.isFunction( obj ) || jQuery.isWindow( obj ) ) {
		return false;
	}
	return type === "array" || length === 0 ||
		typeof length === "number" && length > 0 && ( length - 1 ) in obj; // `( length - 1 ) in obj` checked
}
```
- `underscore`
```javascript
  var isArrayLike = function(collection) {
    var length = getLength(collection);
    return typeof length == 'number' && length >= 0 && length <= MAX_ARRAY_INDEX; // not check
 };
```